### PR TITLE
feat: pipeline orchestrator and RunTransactionAnalysisJob (#162)

### LINKED ISSUE #162: Pipeline orchestrator and RunTransactionAnalysisJob
## Description

Create the `TransactionAnalysisPipeline` orchestrator service and the `RunTransactionAnalysisJob` that dispatches it. The orchestrator loops through registered pipeline stages, handles sequencing, error isolation, and audit logging.

**Parent Epic:** #160
**Depends on:** #161 (Foundation)

## TransactionAnalysisPipeline Service

**File:** `app/Services/TransactionAnalysisPipeline.php`

Constructor takes `array $stages` (list of `PipelineStageContract` implementations).

`run(User $user, PipelineTrigger $trigger): PipelineRun` method:
1. Determine `isFirstSync`: `$user->primary_account_id === null && !$user->hasPayCycleConfigured()`
2. Create `PipelineRun` record with status `Running`
3. Supersede previous pending suggestions for this user (set status to `Superseded`)
4. Build `PipelineContext` DTO
5. Loop stages: check `shouldRun()`, wrap `execute()` in try/catch, track completed/skipped/failed
6. Set overall status: `Completed` (all ok) | `PartialFailure` (some failed) | `Failed` (all failed)
7. Update run with `completed_at` and status
8. Return `PipelineRun`

### Error Isolation
Each stage failure is caught, logged to `stages_failed` and `PipelineAuditEntry`, then the orchestrator continues to the next stage. One stage failing does not abort the pipeline.

### Registration

In `AppServiceProvider::register()`:
```php
$this->app->singleton(TransactionAnalysisPipeline::class, fn () => new TransactionAnalysisPipeline(
    stages: [
        new IdentifyPrimaryAccountStage(),
        new SetPayCycleStage(),
        new IdentifyRecurringTransactionsStage(),
    ],
));
```

## RunTransactionAnalysisJob

**File:** `app/Jobs/RunTransactionAnalysisJob.php`

- `ShouldQueue`, `ShouldBeUnique`
- `tries = 1`, `timeout = 300`
- `WithoutOverlapping("analysis-user-{$this->user->id}")`
- `uniqueId()` returns user ID
- `handle()` resolves `TransactionAnalysisPipeline` from container, calls `run()`
- `failed()` logs error

Follow existing job patterns from `SyncTransactionsJob`.

## Tasks

- [ ] Create `TransactionAnalysisPipeline` service
- [ ] Register as singleton in `AppServiceProvider`
- [ ] Create `RunTransactionAnalysisJob`
- [ ] Write orchestrator tests (mock stages to verify sequencing, error handling, superseding, audit logging)
- [ ] Write job tests (uniqueness, overlap prevention)
- [ ] Run `op ci` to verify

## Acceptance Criteria

- Orchestrator runs stages in order, passes shared context
- Failed stages are isolated — pipeline continues to next stage
- Previous pending suggestions are superseded on new run
- Audit entries created for each stage action
- Job follows existing patterns (ShouldBeUnique, WithoutOverlapping)
- All tests pass, `op ci` green

### PR DESCRIPTION
**Pull request by robwilde**
## Summary

- Implements `TransactionAnalysisPipeline` orchestrator service that loops through registered `PipelineStageContract` stages with per-stage error isolation, status tracking, and audit logging
- Adds `RunTransactionAnalysisJob` queued job following `SyncTransactionsJob` pattern with `ShouldBeUnique`, `WithoutOverlapping`, `tries=1`, `timeout=300`
- Registers `TransactionAnalysisPipeline` singleton in `AppServiceProvider` with empty stages array (concrete stages are future tickets #163+)

## Details

**Orchestrator (`TransactionAnalysisPipeline`):**
- Creates `PipelineRun` record, supersedes pending suggestions, loops stages with try/catch isolation
- Determines `isFirstSync` from user's `primary_account_id` and `hasPayCycleConfigured()`
- Final status derived from completed/failed arrays: `Completed`, `PartialFailure`, or `Failed`
- Creates `PipelineAuditEntry` for each stage outcome

**Job (`RunTransactionAnalysisJob`):**
- Per-user uniqueness via `ShouldBeUnique` + `WithoutOverlapping`
- `tries=1` (no retries — each run creates a PipelineRun, retrying would create duplicates)
- Hardcodes `PipelineTrigger::Sync` (manual trigger comes from a different entry point)

## Test plan

- [x] 16 feature tests for `TransactionAnalysisPipeline` covering: status determination, error isolation, stage skipping, audit entries, suggestion superseding, `isFirstSync` logic, and empty stages edge case
- [x] 5 feature tests for `RunTransactionAnalysisJob` covering: uniqueness, middleware, configuration, handle delegation, and failure logging
- [x] `op lint.dirty` — Pint clean
- [x] `op ci` — full CI green (1086 passed, PHPStan clean)

Closes #162

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/Services/TransactionAnalysisPipeline.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            try {
+                $result = $stage->execute($context);
+                $stagesCompleted[] = $stage->key();
+
+                PipelineAuditEntry::create([
+                    'pipeline_run_id' => $pipelineRun->id,
+                    'stage' => $stage->key(),
+                    'action' => 'completed',
+                    'metadata' => ['suggestion_ids' => $result->suggestionIds],
+                ]);
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/TransactionAnalysisPipeline.php]: Fixed. The pipeline now checks `StageResult::$success` after `execute()` — when `false`, the stage is routed to `stages_failed` with the appropriate audit entry and log. Three new tests cover this path: graceful failure recording, audit entry creation, and mixed with passing stages for `PartialFailure` status.

File: app/Services/TransactionAnalysisPipeline.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        $stagesCompleted = [];
+        $stagesSkipped = [];
+        $stagesFailed = [];
+
+        foreach ($this->stages as $stage) {
+            if (! $stage->shouldRun($context)) {
+                $stagesSkipped[] = $stage->key();
+
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/TransactionAnalysisPipeline.php:64]: Intentionally not adding audit entries for skipped stages. Skips are already tracked in the `stages_skipped` array on the `PipelineRun` model, and the plan deliberately only specifies audit entries for `completed` and `failed` outcomes. Skipping isn't operationally interesting enough to warrant individual audit records.

File: app/Services/TransactionAnalysisPipeline.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    'pipeline_run_id' => $pipelineRun->id,
+                    'stage' => $stage->key(),
+                    'action' => 'failed',
+                    'metadata' => ['error' => $e->getMessage()],
+                ]);
+
+                Log::error('Pipeline stage failed', [
+                    'stage' => $stage->key(),
+                    'pipelineRunId' => $pipelineRun->id,
+                    'error' => $e->getMessage(),
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/TransactionAnalysisPipeline.php:109]: Fixed. Added `'exception' => $e` to the catch block's `Log::error` context, matching the `SyncTransactionsJob::failed()` convention so stack traces are captured.

File: tests/Feature/Services/TransactionAnalysisPipelineTest.php [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        public function shouldRun(PipelineContext $context): bool
+        {
+            return $this->runs;
+        }
+
+        public function execute(PipelineContext $context): StageResult
+        {
+            return new StageResult(success: true, stage: $this->stageKey, suggestionIds: [1, 2]);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (robwilde) [reply tests/Feature/Services/TransactionAnalysisPipelineTest.php:48]: Added. Three new tests cover the `StageResult(success: false)` path:
- `records stage as failed when StageResult returns success false`
- `creates failed audit entry for graceful StageResult failure`
- `mixes graceful failures with passing stages for PartialFailure status`

Skipped stage audit entries intentionally omitted (see reply on the other comment).
